### ISSUE TITLE
RF4.0.1: robot.libdoc no longer processes resource files unless full pathname is given

### ISSUE THREAD

**Issue by oboehmer** [robotframework/robotframework#3929]
With the RF 4.0.1 upgrade, robot.libdoc is no longer able to process resource files which are not accessible directly, but can  be found using PYTHONPATH. This used to work in 4.0.0 and earlier, and could be a consequence of fixing #3919 

Deails:

```
(lm) OBOEHMER-M-C1EC:~ oboehmer$ robot --version
Robot Framework 4.0.1 (Python 3.7.9 on darwin)
(lm) OBOEHMER-M-C1EC:~ oboehmer$ pwd
/Users/oboehmer
(lm) OBOEHMER-M-C1EC:~ oboehmer$ cat /tmp/resource.robot
*** Keywords ***
Test Kw 1
    No Operation


(lm) OBOEHMER-M-C1EC:~ oboehmer$ PYTHONPATH=/tmp python -m robot.libdoc resource.robot list
Importing library 'resource.robot' failed: Module 'resource' does not contain 'robot'.

Try --help for usage information.

```
it works when I specify the full pathname to the resource file:

```
(lm) OBOEHMER-M-C1EC:~ oboehmer$ PYTHONPATH=/tmp python -m robot.libdoc /tmp/resource.robot list
Test Kw 1
(lm) OBOEHMER-M-C1EC:~ oboehmer$
```

This used to work in RF 4.0.0
```
(lm) OBOEHMER-M-C1EC:~ oboehmer$ robot --version
Robot Framework 4.0 (Python 3.7.9 on darwin)
(lm) OBOEHMER-M-C1EC:~ oboehmer$ PYTHONPATH=/tmp python -m robot.libdoc resource.robot list
Test Kw 1
(lm) OBOEHMER-M-C1EC:~ oboehmer$
```

**Comment by pekkaklarck** [comment]
Yes, this is a regression caused by the fix for #3919. Needs to be fixed on RF 4.0.2. Is that release being planned for May 11 a problem for you?

**Comment by oboehmer** [comment]
> Yes, this is a regression caused by the fix for #3919. Needs to be fixed on RF 4.0.2. Is that release being planned for May 11 a problem for you?

4.0.2 is ok, we can workaround by specifying the full path for now...

**Comment by pekkaklarck** [comment]
Another pretty severe regression found. Probably need to release 4.0.1 sooner than originally planned.
